### PR TITLE
Deprecates /datum/mind/var/master

### PR DESCRIPTION
**Pull request by Mister-Moriarty**
[Internal] [Code Quality]

## About The PR:
Deprecates `/datum/mind/var/master` and `/proc/is_mentally_dominated_by()`, replacing them in most cases with the new `/datum/mind/proc/get_master()`.
Removes superfluous vampire thrall checks in the vampire bite ability code.

## Why Is This needed?
Removal of remnants from the old antagonist system that has been superceded by antagonist datums.

### CODE REVIEW

File: code/datums/gamemodes/spy.dm [hunk 1]
```diff
@@ -111,14 +111,14 @@
 	src.spies.Add(spymind)
 	src.spies[spymind] = leadermind
 	spymind.special_role = "spyminion"
-	spymind.master = leader.ckey
+	//spymind.master = leader.ckey
```
Comment (ZeWaka) [inline code/datums/gamemodes/spy.dm:114]: Much rather this gets removed than commented out and forgotten about

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/antagonist_helpers.dm [hunk 2]
```diff
@@ -17,7 +17,19 @@ proc/get_all_gangs()
...
+ */
+/datum/mind/proc/get_master(antagonist_role_id)
+	if (antagonist_role_id)
+		var/datum/antagonist/subordinate/antagonist_role = src.get_antagonist(antagonist_role_id)
+		if (antagonist_role?.master)
+			. = antagonist_role.master
+		return
+
+	for (var/datum/antagonist/subordinate/antagonist_role in src.antagonists)
+		. = antagonist_role.master // Return the master of the newest antagonist datum in `antagonists`.
```
Comment (ZeWaka) [inline code/modules/antagonists/antagonist_helpers.dm:32]: is the newest at the end or something?

File: code/modules/antagonists/antagonist_helpers.dm [hunk 3]
```diff
@@ -17,7 +17,19 @@ proc/get_all_gangs()
 	for (var/datum/antagonist/gang_leader/antag_datum in get_all_antagonists(ROLE_GANG_LEADER))
 		. += antag_datum.gang
 
-	return .
+/** Returns the mind of this mind's master.
+ * If a specific antagonist role ID is provided, then the master of the antagonist datum of that ID belonging to this mind will be returned.
+ * If no antagonist role ID is provided, then the master of the newest antagonist datum will be returned.
+ */
+/datum/mind/proc/get_master(antagonist_role_id)
+	if (antagonist_role_id)
```
Comment (ZeWaka) [inline code/modules/antagonists/antagonist_helpers.dm:25]: what if this is called with a no id and there is no antagonists? you should set a default return value

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/antagonist_helpers.dm (hunk 2 above)
Comment (Mister-Moriarty) [reply code/modules/antagonists/antagonist_helpers.dm:32]: That is how lists work, to my knowledge. The most recently instantiated antagonist datum _should_ append itself to the end of the `antagonists` list on the owner mind.

File: code/modules/antagonists/antagonist_helpers.dm (hunk 3 above)
Comment (Mister-Moriarty) [reply code/modules/antagonists/antagonist_helpers.dm:25]: Is the default return value not implicitly `null`? If so, to me that seems a sensible value to return if no master could be found.

Comment (ZeWaka) [reply code/modules/antagonists/antagonist_helpers.dm:25]: it is, just wanted to make sure you thought about it

**Pull request merged**